### PR TITLE
feat(downloader): forward headers

### PR DESCRIPTION
**Pull request by kptdobe**
PR for #612

### CODE REVIEW

**Comment by tripodsan** [comment]
^^^ fix commit message `fear` -> `feat`

File: src/utils/Downloader.js [hunk 1]
```diff
@@ -80,6 +86,7 @@ class Downloader {
    * @param {string} opts.id Some id to later identify the task.
    * @param {number} opts.timeout Override global timeout
    * @param {boolean} opts.errorOn404 Treat 404 as error.
+   * @param {boolean} opts.forwardHeaders False to not forward the predefined request headers.
```
Comment (tripodsan) [inline src/utils/Downloader.js]: I would invert the condition. specifying an explicit `false` is counter intuitive. 

or even better, turn in into an array with a default, eg:

```
@param {Array<string>} opts.forwardHeaders Names of headers that are forwarded. Default `['x-request-id', 'x-cdn-request-id', 'x-cdn-url']
```

File: src/utils/Downloader.js [hunk 2]
```diff
@@ -80,6 +86,7 @@ class Downloader {
...
    * @return {Promise} Promise that resolves with the response.
```
Comment (tripodsan) [inline src/utils/Downloader.js]: please add:

```
@param {object} opts.options Fetch options passed to the underling helix-fetch.
```

**Comment by kptdobe** [comment]
@tripodsan thanks for the feedback. I integrated all of them.

File: src/utils/Downloader.js [hunk 3]
```diff
@@ -76,10 +82,12 @@ class Downloader {
    * Schedules a task that fetches a web resource.
    * @param {object} opts options.
    * @param {object} opts.uri URI to download
-   * @param {object} opts.options Fetch options.
+   * @param {object} opts.options Fetch options passed to the underling helix-fetch.
    * @param {string} opts.id Some id to later identify the task.
    * @param {number} opts.timeout Override global timeout
    * @param {boolean} opts.errorOn404 Treat 404 as error.
+   * @param {Array<string>} opts.forwardHeaders Names of headers that are forwarded.
```
Comment (tripodsan) [inline src/utils/Downloader.js]:
```suggestion
   * @param {string[]} opts.forwardHeaders Names of headers that are forwarded.
```
(sorry, this is an easier notation; forgot about it :-)

File: src/utils/Downloader.js [hunk 4]
```diff
@@ -76,10 +82,12 @@ class Downloader {
...
+   * Default `['x-request-id', 'x-cdn-request-id', 'x-cdn-url']
```
Comment (tripodsan) [inline src/utils/Downloader.js]:
```suggestion
   * Default `['x-request-id', 'x-cdn-request-id', 'x-cdn-url']`
```

**Comment by kptdobe** [comment]
I still need a second reviewer.

**Pull request merged**